### PR TITLE
Statically build using musl toolchain and target alpine

### PR DESCRIPTION
**Pull request by pjbgf**
Main changes:
- Statically build controller using musl toolchain.
- Simplifies Dockerfile by leveraging the musl toolchain static libraries provided by [golang-with-libgit2@libgit2-1.1.1-4](https://github.com/fluxcd/golang-with-libgit2/releases/tag/libgit2-1.1.1-4).
- Target `alpine` as final base image.
- Add file containing attributions for libraries that are statically built into controller.
- Add `make verify`.

Although the controller is statically built, we are targeting `alpine` to help users debugging the image if they so need/wish.

### CODE REVIEW

**Comment by pjbgf** [comment]
Needs to be updated with source controller once https://github.com/fluxcd/source-controller/pull/558 is merged.

**Review by relu — APPROVED** [review]
Looks solid to me. :heavy_check_mark:

File: Dockerfile [hunk 1]
```diff
@@ -47,45 +26,47 @@ COPY go.sum go.sum
...
-RUN xx-go build -o image-automation-controller -trimpath \
+
+RUN xx-apk add --no-cache \
+        musl-dev gcc lld binutils-gold
+
+# Performance related changes:
+# - Use read-only bind instead of copying go source files.
+# - Cache go packages.
+RUN --mount=target=. \
+    --mount=type=cache,target=/root/.cache/go-build \
```
[the remark this answers is not in the document]
Comment (pjbgf) [reply Dockerfile:41]: Thank you @relu, I will take a look and check ways to improve it. :+1:

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @pjbgf 🏅 

I've tested this on several EKS ARM64 clusters where the previous release was failing 90% of the time with `✗ ImageUpdateAutomation reconciliation failed: 'unable to clone: SSH could not read data: Error waiting on socket'`. After using a build from this PR all errors were gone 🎉

**Review by hiddeco — APPROVED** [review]
Updated the branch protection rules so this one can pass, thanks @pjbgf :bow:

**Pull request merged**